### PR TITLE
opp-association

### PR DESCRIPTION
**Pull request by ger619**
- Create a class Classroom with the following:
    - @label instance variable, should be initialized in the constructor.
    - Setter and getter for @label (remember about attr_accessor).
- Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
    - Create the has-many side (a classroom has many students).
    - Create the belongs-to side (a student belongs to a classroom).
    - Make sure that when adding a student to a classroom it also sets the classroom for the student.
    - Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Create a class Book with the following:
  - @title and @author instance variables, should be initialized in the constructor.
  - Setters and getters for instance variables (remember about attr_accessor).
- Create a class Rental with the following:
- @date instance variable, should be initialized in the constructor.
- Setter and getter for @date (remember about attr_accessor).
- Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
  - Create the has-many side of Book and Rental (a book has many rentals).
  - Create the belongs-to side of Rental and Book (a rental belongs to a book).
  - Create the has-many side of Person and Rental (a person has many rentals).
  - Create the belongs-to side of Rental and Person (a rental belongs to a person).
  - Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -20,6 +20,10 @@ def correct_name
     @name
   end
```
Comment (danmainah) [inline person.rb:23]: - [ ] @ger619 great work, you forgot to initialize @rentals and assign it an empty array, also you need to add rentals to your attr_accessor.

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+require_relative './rental'
+class Book
+  attr_accessor :title, :author
+  attr_reader :rental
```
Comment (danmainah) [inline book.rb]: - [ ] @ger619 the rental should have access to read and write like the title and author, kindly move it to attr_accessor.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @ger619 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Github flow was correctly used


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: person.rb [hunk 3]
```diff
@@ -20,6 +20,14 @@ def correct_name
...
 
+  def rentals
+    @rental
+  end
```
Comment (DammyShittu) [inline person.rb]: - [ ] Definition of a new method is not what the previous reviewer meant. The previous reviewer meant you should add an empty rentals variable to your `initialize` constructor. See example below. Kindly remove this method as it is unnecessary.

  ```
  def initialize(age, name = 'unknown', parent_permission: true)
    super()
    @id = Random.rand(1..1000)
    @age = age
    @name = name
    @parent_permission = parent_permission
    @rentals = []
  end
  ```

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @ger619,

## Highlights
✔️ Class relationships
✔️ Descriptive PR

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by descholar-ceo — APPROVED** [review]
Hi @ger619 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
👍🏽 Implemented the requested changes ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**